### DISCUSSION TITLE
Load font from non-seekable stream

### DISCUSSION THREAD

**Discussion by jahav** [SixLabors/Fonts#319] [Ideas] [answered]
A lot of environments today don't provide reliable access to file system. E.g. Client side Blazor, Azure Functions ect.

In such cases, it would be greate to load font from a some web resource, but these streams are in many cases nonseekable. Fonts can't load font from nonseekable stream, it throws at `TableHeader.CreateReader` when it attemps to see at the beginning of the table header and would likely throw on other places.

Workaround: It is possible to read whole non-seekable stream to an intermediate `MemoryStream` and use the MS instead.

**Comment by tocsoft** [comment, marked as answer]
The way the font file format works is it effectivly requires processing the font data in a random access fashion, so even if we supported using non seekable streams all we would be able to do to 'fix' it would be do the copy silently anyway.

**Reply by jahav** [reply]
I see. Thank you for consideration and explanation.
